lib/agent0/agent0/hyperdrive/exec/run_agents.py

```diff
@@ -380,3 +391,99 @@ def build_wallet_positions_from_data(
         longs=long_obj,
         shorts=short_obj,
     )
+
+
+def build_wallet_positions_from_chain(
+    wallet_addr: str, hyperdrive_contract: IERC4626HyperdriveContract, base_contract: ERC20MintableContract
+) -> HyperdriveWallet:
+    """Builds a wallet position based on gathered data.
+
+    Arguments
+    ---------
+    wallet_addr: str
+        The checksum wallet address
+    hyperdrive_contract: Contract
+        The Hyperdrive contract to query the data from
+    base_contract: Contract
+        The base contract to query the base amount from
+
+    Returns
+    -------
+    HyperdriveWallet
+        The wallet object build from the provided data
+    """
+    # Contract call to get base balance
+    base_amount: int = base_contract.functions.balanceOf(wallet_addr).call()
+    # TODO do we need to do error checking here?
+    base_obj = Quantity(amount=FixedPoint(scaled_value=base_amount), unit=TokenType.BASE)
+
+    # Contract call to get lp balance
+    asset_id = encode_asset_id(AssetIdPrefix.LP, 0)
+    lp_amount: int = hyperdrive_contract.functions.balanceOf(asset_id, wallet_addr).call()
+    lp_obj = FixedPoint(scaled_value=lp_amount)
+
+    # Contract call to get withdrawal positions
+    asset_id = encode_asset_id(AssetIdPrefix.WITHDRAWAL_SHARE, 0)
+    withdraw_amount: int = hyperdrive_contract.functions.balanceOf(asset_id, wallet_addr).call()
+    withdraw_obj = FixedPoint(scaled_value=withdraw_amount)
+
+    # We need to gather all longs and shorts from events
+    # and rebuild the current long/short positions
+    # Open Longs
+    open_long_events = hyperdrive_contract.events.OpenLong.get_logs(fromBlock=0)
```

Comment:
my understanding from our earlier convo about this is that it is not guaranteed that we can get all events from block 0. I think we should wrap these in a try/except that throws a more detailed error if it fails (e.g. "Unable to load events from block 0, are you using a node that stores all historic data?")

Reply:
My takeaway from this is that we're always to get logs regardless of the chain, it just might be slow.

Reply:
I don't know how to query what the oldest available block is... if that were possible we could have an alternate flow that gets the earliest possible and posts a warning? The user might not care that we can't access block 0 if the bot is relatively new.

Reply:
> My takeaway from this is that we're always to get logs regardless of the chain, it just might be slow.

I'm not sure about that, will have to verify.  I think it depends on your provider, which changes with the rpc url given to web3.

Reply:
even our local devnet with anvil sometimes fails to give us block 0, right?

Reply:
even if your takeaway is correct we can go with my original proposal (try/except) as a jic, and then put a TODO to more cleverly use earliest available block in a new PR

Reply:
We should revisit this discussion, logs are always available (may be slow), its only if we want to get data from blocks via an anvil specific thing is where it fails. If logs are not always available, then we have bigger problems because frontend depends on this process as well.

Reply:
I don't think we need a try/catch here, we probably want the code to crash if this isn't the case (again, we have a huge issue in frontend if this isn't the case), hence, I'd rather not catch exceptions and have execution fail.